ships/0004-build-env-vars.md

```diff
@@ -0,0 +1,241 @@
+
+At the same time, build stategy authors may want to control which environment variables can be altered by the end developer.
+Many tools use environment variables to override basic security settings, like TLS verification, that are otherwise enabled by default.
+A build strategy author may not want developers to change these settings, or would perfer to provide default values that adhere to best practices.
+
+### Goals
+
+* Allow developers to set environment variables used in a build step.
+* Provide a means for build strategy authors to keep fixed values for environment variables.
+
+### Non-Goals
+
+* Allow developers to alter or override other aspects of the build step, such as commands.
+* Set environment variables in the output image of the build.
+* Validate environment variables against those in the base image of a build step.
+* Provide direct support for Dockerfile [build arguments](https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables---build-arg).
+
+## Proposal
+
+### User Stories
+
+#### Story: Alter build behavior via environment variables
+
+As a nodejs developer using buildpacks
+I want to set the `NPM_INCLUDE` environment variable in my Build and BuildRuns
+So that I can include development and debug resources in my container image
+
+#### Story: Prevent insecure behavior in a build strategy
+
+As a build strategy author
+I want to keep fixed values for some environment variables in my build strategy
+So that I don't allow end users to enable insecure or potentially harmful features.
+
+#### Story: Provide credentials via environment variables
+
+As a Java developer using maven to build my application (via buildpacks or source-to-image)
+I want to use environment variables to pass usernames and passwords to values defined in my `pom.xml` file
+So that I don't store sensitive information in source control.
+
+### Implementation Notes
+
+#### Build Strategy Behavior
+
+Build strategies which declare an environment variable in a build strategy step will by default be fixed.
+A developer will not be able to change this value directly.
+
+If a build strategy author wants to allow an environment variable to be tuned, they have two options:
+
+1. Omit the environment variable - this works if an empty/zero default value is acceptable in the build step.
+2. Declare the envionment variable, but use a build strategy parameter to set the value.
+   This is useful if the build strategy should use a nonempty/nonzero default value in the build step.
+
+For example, a BuildKit build strategy author could use the following to ensure that TLS is always verified when pushing the resulting image:
+
+```yaml
+spec:
+  steps:
+  - name: build
+    ...
+    env:
+    - name: DOCKER_TLS_VERIFY
+      value: "true"
+```
+
+Or alternatively, the build strategy author can make this configurable with a build parameter:
+
+```yaml
+spec:
+  params:
```

Comment:
we change this one, see https://github.com/shipwright-io/build/blob/main/docs/buildstrategies.md#strategy-parameters . Can you use here `parameters` pls.

Reply:
done.